### PR TITLE
chore(deps): add workflow to automatically update enarx dependencies

### LINKED ISSUE #1486: [Infrastructure]: Update enarx dependencies weekly
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Description

Provide automation for the weekly update of `cargo` crate dependencies, test and create PR on successful test pass.

### PR DESCRIPTION
**Pull request by platten**
Closes #1486 
Signed-off-by: Paul Pietkiewicz <paul@profian.com>

### CODE REVIEW

**Review by bstrie — DISMISSED** [review]

File: .github/workflows/cargo-update.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  cancel-in-progress: true
+name: cargo-update
+jobs:
+  create-pr:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2.4.0
+      - name: Update dependencies
+        run: helper/bump-dependencies.sh 
+        shell: bash
```
Comment (npmccallum) [inline .github/workflows/cargo-update.yml:17]: Let's avoid a helper script here. Just call `cargo update` four times with the appropriate `--manifest-path=$PATH/Cargo.toml` argument.

**Review by npmccallum — CHANGES_REQUESTED** [review]

File: .github/workflows/cargo-update.yml (hunk 1 above)
Comment (platten) [reply .github/workflows/cargo-update.yml:21]: Addressed

**Comment by platten** [comment]
> please rebase for the new CI tests

Done!

File: .github/workflows/cargo-update.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+jobs:
+  create-pr:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2.4.0
+      - name: Update dependencies
+        run: |
+         cargo update --manifest-path=internal/shim-kvm/Cargo.toml
+         cargo update --manifest-path=internal/shim-sgx/Cargo.toml
+         cargo update --manifest-path=internal/wasmldr/Cargo.toml
```
Comment (haraldh) [inline .github/workflows/cargo-update.yml:19]:
```suggestion
         cargo update --manifest-path=src/bin/wasmldr/Cargo.toml
```

**Review by haraldh — CHANGES_REQUESTED** [review]

**Review by haraldh — CHANGES_REQUESTED** [review]

File: .github/workflows/cargo-update.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          labels: dependencies
+          signoff: true
+          branch: chore/cargo-update
+          head: ${{ github.ref }}
+          base: main
+          add-paths: |
+            Cargo.lock
+            internal/shim-sev/Cargo.lock
+            internal/shim-sgx/Cargo.lock
+            internal/wasmldr/Cargo.lock
```
[the remark this answers is not in the document]
Comment (platten) [reply .github/workflows/cargo-update.yml]: Fixed. I think we should be good!

**Review by haraldh — APPROVED** [review]

File: .github/workflows/cargo-update.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+          labels: dependencies
+          signoff: true
+          branch: chore/cargo-update
+          head: ${{ github.ref }}
+          base: main
+          add-paths: |
+            Cargo.lock
+            internal/shim-sev/Cargo.lock
+            internal/shim-sgx/Cargo.lock
+            src/bin/wasmldr/Cargo.lock
```
Comment (haraldh) [inline .github/workflows/cargo-update.yml:38]: This path does not exist anyway, I figured!

Comment (haraldh) [reply .github/workflows/cargo-update.yml:38]: `src/bin/wasmldr/Cargo.lock` I mean

**Pull request merged**